hbase-common/src/main/java/org/apache/hadoop/hbase/util/CommonFSUtils.java

```diff
@@ -759,200 +739,75 @@ public static void checkShortCircuitReadBufferSize(final Configuration conf) {
     conf.setIfUnset(dfsKey, Integer.toString(hbaseSize));
   }
 
-  private static class DfsBuilderUtility {
-    static Class<?> dfsClass = null;
-    static Method createMethod;
-    static Method overwriteMethod;
-    static Method bufferSizeMethod;
-    static Method blockSizeMethod;
-    static Method recursiveMethod;
-    static Method replicateMethod;
-    static Method replicationMethod;
-    static Method buildMethod;
-    static boolean allMethodsPresent = false;
+  private static final class DfsBuilderUtility {
+    private static final Class<?> BUILDER;
+    private static final Method REPLICATE;
 
     static {
-      String dfsName = "org.apache.hadoop.hdfs.DistributedFileSystem";
-      String builderName = dfsName + "$HdfsDataOutputStreamBuilder";
+      String builderName = "org.apache.hadoop.hdfs.DistributedFileSystem$HdfsDataOutputStreamBuilder";
```

Comment:
`DistributedFileSystem` appears to exist since 2.10, so can this string be built from a class reference? For example,

```java
String buildName = DistributedFileSystem.class.getName() + "$HdfsDataOutputStreamBuilder";
```

Looks like hbase-common module doesn't depend on hadoop-hdfs-client jar, so maybe this gets too weird for a "cleanup" patch. Would be nice though, to use whatever static references we can...

Reply:
Yes, hbase-common only depends on hadoop-common so we can only access FileSystem, not DistributedFileSystem. I tried to move this method to hbase-server but then I realized that we also use this method in hbase-procedure, so leave it as is for now. The WALProcedureStore is supposed to be removed in 4.0.0, let's revive this again at that time...